src/main/java/org/assertj/core/api/SoftAssertion.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.util.List;
+
+public interface SoftAssertion {
+	<V extends Assert<? extends V, ? extends T>, T> V proxy(Class<V> assertClass, Class<T> actualClass, T actual);
```

Comment:
Done. I'm also going to put this change (and related) into a separate commit, because adding bounds to these type parameters is not strictly part of the soft assertions restructure.